### PR TITLE
Feat: Byline links

### PR DESCRIPTION
**Pull request by aronshamash**
As per: https://nidigitalsolutions.jira.com/browse/REPLAT-2735

- Bylines are not clickable on article list pages
- Bylines are grey on article list pages
- Bylines are coloured on article pages by section
- Bylines are are underlined on article pages when linkable
- Bylines are coloured in related articles by section

### CODE REVIEW

File: packages/article-byline/article-byline-with-links.showcase.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  ]);
+
+const bylineLinkStyles = {
+  link: {
+    color: colours.functional.action,
+    textDecorationLine: "none"
+  }
+};
+
+const getCommonProps = (selectV2, decorateAction) => ({
```
Comment (colincclark) [inline packages/article-byline/article-byline-with-links.showcase.js]: `getProps()` is better

File: packages/article-byline/article-byline-with-links.showcase.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+const bylineLinkStyles = {
+  link: {
+    color: colours.functional.action,
+    textDecorationLine: "none"
+  }
+};
+
+const getProps = (selectV2, decorateAction) => ({
+  /* We're using selectV2 over our own select abstraction as
```
Comment (jimhunty) [inline packages/article-byline/article-byline-with-links.showcase.js]: This sentance doesnt make sense.

File: packages/article-byline/article-byline-with-links.showcase.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
+/* eslint-disable react/prop-types */
+
+import React from "react";
+import { Text } from "react-native";
+import { fonts, colours, fontSizes } from "@times-components/styleguide";
+import { ArticleBylineWithLinks } from "./src/article-byline";
+
+const authorsAST = require("./fixtures/authors.json");
+
+const bylineStyles = {
```
Comment (jimhunty) [inline packages/article-byline/article-byline-with-links.showcase.js]: Worth importing from elsewhere? Like in the byline component if it has a default style?

File: packages/article-byline/src/article-byline-prop-types.js [hunk 4]
```diff
@@ -7,11 +7,13 @@ export const articleBylinePropTypes = {
...
-  })
+  }),
+  color: PropTypes.string
 };
 
 export const articleBylineDefaultPropTypes = {
   ast: {},
   onAuthorPress: () => null,
-  style: {}
+  style: {},
```
Comment (jimhunty) [inline packages/article-byline/src/article-byline-prop-types.js]: Is it worth having better defaults here? How does it look without them?

File: packages/article-byline/src/article-byline.js [hunk 5]
```diff
@@ -1,51 +1,24 @@
...
+  const bylineStyles = getBylineStyles(style, color);
+  return renderByline(renderAuthorComponent, ast, bylineStyles);
 };
 
 ArticleByline.propTypes = articleBylinePropTypes;
 ArticleByline.defaultProps = articleBylineDefaultPropTypes;
 
-export { articleBylinePropTypes };
 export default ArticleByline;
+export { default as ArticleBylineWithLinks } from "./article-byline-with-links";
```
Comment (jimhunty) [inline packages/article-byline/src/article-byline.js]: I prefer default export below.

File: packages/article-summary/article-summary.showcase.js [hunk 6]
```diff
@@ -21,9 +21,15 @@ export default {
...
       type: "story",
-      name: "With links",
+      name: "With links in summary",
       component: () => story(<ArticleSummary {...withLinksFixture} />)
     },
+    {
+      type: "story",
+      name: "With byline links",
+      component: () =>
+        story(<ArticleSummary {...defaultFixture} hasBylineLinks />)
```
Comment (jimhunty) [inline packages/article-summary/article-summary.showcase.js]: Article Summary takes in an object of byline props. `hasBylineLinks ` should exist in there.

File: packages/article-byline/src/article-byline-prop-types.js (hunk 4 above)
Comment (aronshamash) [reply packages/article-byline/src/article-byline-prop-types.js]: These are the styles added in showcase:
```
  fontSize: fontSizes.cardMeta,
  fontFamily: fonts.supporting,
  color: colours.functional.secondary,
  lineHeight: 13,
  flexDirection: "row"
```

I guess they could be made defaults but would want to confirm with Jake.

File: packages/article-byline/src/article-byline-with-links.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+      onPress={e => onAuthorPress(e, { slug: attributes.slug, url })}
+    >
+      {children}
+    </TextLink>
+  );
+};
+
+const ArticleBylineWithLinks = ({ ast, style, color, onAuthorPress }) => {
+  const bylineStyles = getBylineStyles(style, color);
+  return renderByline(renderAuthorComponent, ast, bylineStyles, onAuthorPress);
```
Comment (craigbilner) [inline packages/article-byline/src/article-byline-with-links.js]: minor: as a rule you want fixed/static args at the front and the "variable" at the end should you need to refactor into a curried function/thunk

Comment (craigbilner) [reply packages/article-byline/src/article-byline-with-links.js]: also not sure `bylineStyles` brings too much to the party

File: packages/article/__tests__/android/article.android.test.js [hunk 8]
```diff
@@ -2,7 +2,10 @@ import shared from "../shared";
 import nativeShared from "../shared-native";
 import sharedTracking from "../shared-tracking";
 
-jest.mock("@times-components/article-byline", () => "ArticleByline");
+jest.mock("@times-components/article-byline", () =>
+  // eslint-disable-next-line global-require
+  require("../articleBylineMock")
```
Comment (craigbilner) [inline packages/article/__tests__/android/article.android.test.js]: does this need to be dynamic, is it not fixed

*file needs to be lower kebab case*

File: packages/article-byline/src/article-byline-prop-types.js (hunk 4 above)
Comment (jimhunty) [reply packages/article-byline/src/article-byline-prop-types.js]: I think make these defaults for now and then confirm them in design review. Should have a default story as well to make this visible

Comment (aronshamash) [reply packages/article-byline/src/article-byline-prop-types.js]: taking out of showcase and made these the default which are visible in all the stories

**Pull request merged**